src/interfaces/wasi/wit_implementation.rs

```diff
@@ -0,0 +1,88 @@
+wit_bindgen_rust::export!("abe.wit");
+struct Abe;
+
+use abe::*;
+
+use super::wit_generation;
+
+impl abe::Abe for Abe {
+    fn generate_master_key(nb_revocation: u64, policy: Policy) -> Result<MasterKey, String> {
```

Comment:
take by ref if you can (other cases later on)
```suggestion
    fn generate_master_key(nb_revocation: u64, policy: &Policy) -> Result<MasterKey, String> {
```

Reply:
actually here we cannot  take parameters by reference. It is du to WIT syntax which cannot not support references (for the time being at least)